jvector-base/src/main/java/io/github/jbellis/jvector/graph/NodeSimilarity.java

```diff
@@ -70,6 +74,13 @@ default boolean isExact() {
     }
 
     interface Reranker {
-        float similarityTo(int node2);
+        /**
+         * Rerank the nodes identified by the nodes array, storing the similarities in the results vector.
+         *
+         * @param nodes   the nodes to calculate similarity for
+         * @param results the similarity scores for each node, in the same order as the input nodes.
+         *                This should be pre-allocated to the same size as nodes.length.
+         */
+        void similarityTo(int[] nodes, VectorFloat<?> results);
```

Comment:
let's rename this, the original name implies in my mind that it's returning the similarity